src/SW_netCDF.c

```diff
@@ -0,0 +1,591 @@
+    int yDimID, xDimID, bndsDimID, siteDimID;
+
+    nc_create(DomainName, NC_NETCDF4, NULL); // Don't store file ID (OPEN_NC_ID)
+
+    create_var_crs(LogInfo);
+    write_global_domain_atts(LogInfo);
+
+    if(strcmp(SW_Domain->DomainType, "xy") == equalString) {
+        create_dim_xy(SW_Domain->nDimX, SW_Domain->nDimY, &yDimID,
+                      &xDimID, &bndsDimID, LogInfo);
+        create_xy_vars(SW_Domain->nDimX, SW_Domain->nDimY, yDimID,
+                       xDimID, bndsDimID, LogInfo);
+    } else {
+        create_dim_s(SW_Domain->nDimS, &siteDimID, LogInfo);
+        create_s_vars(SW_Domain->nDimS, siteDimID, LogInfo);
+    }
+
+    nc_close(OPEN_NC_ID);
+}
+
+/**
+ * @brief Creates dimensions "x", "y", and "bnds" for the domain type "xy"
+ *
+ * @param[in] nDimX Size of dimension "x"
+ * @param[in] nDimY Size of dimension "y"
+ * @param[in] yDimID Dimension ID of "y" within the domain
+ * @param[in] xDimID Dimension ID of "x" within the domain
+ * @param[in] bndsDimID Dimension ID of "bnds" within the domain
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void create_dim_xy(int nDimX, int nDimY, int* yDimID, int* xDimID,
+                   int* bndsDimID, LOG_INFO* LogInfo) {
+
+    // Create dimensions "x", "y", and "bnds" and don't store variable ID
+    create_dimension("x", xDimID, nDimX, LogInfo);
+    create_dimension("y", yDimID, nDimY, LogInfo);
+    create_dimension("bnds", bndsDimID, NUMBNDS, LogInfo);
+}
+
+/**
+ * @brief Creates dimension "site" for the domain type "s" (sites)
+ *
+ * @param[in] nDimS Size of dimension "site"
+ * @param[in] sDimID Dimension ID of "site" within the domain
+ *
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void create_dim_s(int nDimS, int* sDimID, LOG_INFO* LogInfo) {
+
+    // Create dimension "site" and don't store variable ID
+    create_dimension("site", sDimID, nDimS, LogInfo);
+}
+
+/**
+ * @brief Creates/files all variables related to domain "xy"
+ *  in the new domain netCDF
+ *
+ * @param[in] nDimX Size of dimension "x"
+ * @param[in] nDimY Size of dimension "y"
+ * @param[in] yDimID Dimension ID of "y" with the domain
+ * @param[in] xDimID Dimension ID of "x" with the domain
+ * @param[in] bndsDimID Dimension ID of "bnds" with the domain
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void create_xy_vars(int nDimX, int nDimY, int yDimID, int xDimID,
+                    int bndsDimID, LOG_INFO* LogInfo) {
+
+    int xDim[] = {xDimID}, yDim[] = {yDimID}, xBndsDim[] = {xDimID, bndsDimID},
+        yBndsDim[] = {yDimID, bndsDimID}, domainDims[] = {yDimID, xDimID};
+    int xID, yID, xBndsID, yBndsID, domID;
+    size_t numChunkVals = 2;
+    unsigned int xBndsAttVals[] = {nDimX, NUMBNDS},
+        yBndsAttVals[] = {nDimY, NUMBNDS}, ChunkVals[] = {nDimX, nDimY};
+    float fillVal[] = {-3.4E38};
+
+    // "domain" variable
+    create_var_domain(&domID, domainDims, TWODIMS, numChunkVals,
```

Comment:
I'm not sure I understand the problem in this comment. The variable `domain` is created in these functions because when creating the variable, mainly, we can set domain-specific attributes.

I'm not sure I see what exactly, aside from `domain` itself, cannot be used outside of a domain environment (ignoring other comments). Because we have `create_variable()` and `create_dimension()` along with others which are generic. When we use `create_*_vars()` it is meant to create the entire domain file based on the domain type. With helper functions that should be generic enough to use in other occasions. Though `create_var_*()` may need changes depending on the expected attributes for output variable files.